app/controllers/admin_controller.rb

```diff
@@ -0,0 +1,25 @@
+class AdminController < ActionController::Base
+  def index 
+    @all_applications = PetsApplication.all
+  end
+  def shelters_index
+    @shelters = Shelter.order_by_name_alphabetical
+    @pending_apps = Shelter.with_pending_applications
+  end
+
+  def show
+    @application = PetsApplication.find(params[:id])
+    @applicant = Applicant.find(@application.applicant_id)
+    @pets = Pet.pets_with_app_status_by_sql(@application)
+    @status = PetsApplication.check_app_status(@applicant)
+  end
+
+  def approve_reject
```

Comment:
I am wondering if this could live on your `update` action in your `pet_applications` controller??